File: Day19_LetterComb.py
def letterCombinations(self, digits: str) -> list[str]:
    if(not len(digits)):
        return []
    m = {
        '2' : "abc",
        '3' : "def",
        '4' : "ghi",
        '5' : "jkl",
        '6' : "mno",
        '7' : "pqrs",
        '8' : "tuv",
        '9' : "wxyz",
    }
    ans = []
    def back(s, s1, m):
        if(s == ""):
            ans.append(s1)
            return
        for c in m[s[0]]:
            back(s[1:], s1+c, m)
    back(digits, "", m)
    return ans

File: Day19_LetterComb.cpp
#include <bits/stdc++.h>
using namespace std;

vector<string> ans;
void back(string s, string s1, map<char, string> m)
{
    if (s == "")
    {
        ans.push_back(s1);
        return;
    }
    for (auto &c : m[s[0]])
    {
        string s2 = s;
        s2.erase(s2.begin());
        back(s2, s1 + c, m);
    }
}

vector<string> letterCombinations(string digits)
{
    if (digits.length() == 0)
    {
        vector<string> v = {};
        return v;
    }
    map<char, string> m;
    m['2'] = "abc";
    m['3'] = "def";
    m['4'] = "ghi";
    m['5'] = "jkl";
    m['6'] = "mno";
    m['7'] = "pqrs";
    m['8'] = "tuv";
    m['9'] = "wxyz";
    back(digits, "", m);
    return ans;
}

int main()
{

    return 0;
}